ocaml/xenopsd/lib/topology.ml

```diff
@@ -194,25 +187,19 @@ module NUMA = struct
                None
          )
     in
-    let single_nodes =
-      valid_nodes
-      |> Seq.map (fun i ->
-             let self_distance = d.(i).(i) in
-             (distance_to_candidate self_distance, Seq.return i)
-         )
-    in
     let numa_nodes = Array.length d in
```

Comment:
We could count how many reachable NUMA nodes we have, then we can avoid unreachable nodes artificially triggering the 16 node limit.

We should probably also log when the limit got triggered, so we can debug unexpected cases.

Although that can come as a separate PR, this is already a good improvement.